static/js/pages/instances/actions/FreezeInstanceBtn.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { NotificationHelper } from "types/notification";
+import ConfirmationButton from "components/ConfirmationButton";
+import { Tooltip } from "@canonical/react-components";
+import { freezeInstance, unfreezeInstance } from "api/instances";
+
+interface Props {
+  instance: LxdInstance;
+  notify: NotificationHelper;
+  hasCaption?: boolean;
+  isDense?: boolean;
+}
+
+const FreezeInstanceBtn: FC<Props> = ({
+  instance,
+  notify,
+  hasCaption = true,
+  isDense = true,
+}) => {
+  const [isLoading, setLoading] = useState(false);
+  const queryClient = useQueryClient();
+
+  const handleFreeze = () => {
+    setLoading(true);
+    freezeInstance(instance)
+      .then(() => {
+        setLoading(false);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+        notify.success(`Instance ${instance.name} frozen.`);
+      })
+      .catch((e) => {
+        setLoading(false);
+        notify.failure("Error on instance freeze.", e);
+      });
+  };
+
+  const handleUnfreeze = () => {
+    setLoading(true);
+    unfreezeInstance(instance)
+      .then(() => {
+        setLoading(false);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+        notify.success(`Instance ${instance.name} is now running.`);
+      })
+      .catch((e) => {
+        setLoading(false);
+        notify.failure("Error on instance unfreeze.", e);
+      });
+  };
+
+  const isDisabled =
+    isLoading || instance.status === "Stopped" || instance.status === "Error";
+
+  return (
+    <Tooltip
+      message={isDisabled ? "Instance must be running" : undefined}
+      position="btm-center"
+    >
+      {instance.status !== "Frozen" && (
+        <ConfirmationButton
+          className="u-no-margin--bottom u-no-margin--right"
+          isLoading={isLoading}
+          iconClass={
+            isLoading
+              ? "p-icon--spinner u-animation--spin"
+              : "p-icon--disconnect"
```

Comment:
This looks a bit too much like networking, no?

Reply:
This is the closest I could find within vanilla icon set. The only alternative that comes to my mind is to remove the icon, which would make it look inconsistent with the Start/Stop button (as well as all the other buttons), unless I remove it from there as well.
Any other solution?

Reply:
Let's keep it as is, and I'll make a request for a new icon.